src/ape_geth/provider.py

```diff
@@ -469,4 +471,23 @@ def build_command(self) -> List[str]:
 
 # NOTE: The default behavior of EthereumNodeBehavior assumes geth.
 class Geth(EthereumNodeProvider):
-    pass
+    @property
+    def uri(self) -> str:
+        uri = super().uri
+        ecosystem = self.network.ecosystem.name
+        network = self.network.name
+
+        # If we didn't find one in config, look for a public RPC.
+        if not uri or uri == DEFAULT_SETTINGS["uri"]:
+            # Do not override explicit configuration
+            if ecosystem in self.config:
+                # Shape of this is odd.  Pydantic model containing dicts
```

Comment:
it is not too odd, it isnt much different than using `additonalProperties` in OpenAPI swagger docs

Reply:
Feel free to add a comment under mine disputing my assertion.  :smirk: